private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/maven/MavenResolver.java

```diff
@@ -0,0 +1,423 @@
+import java.util.stream.Collectors;
+import org.apache.maven.repository.internal.MavenRepositorySystemUtils;
+import org.eclipse.aether.DefaultRepositorySystemSession;
+import org.eclipse.aether.RepositoryListener;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.collection.CollectRequest;
+import org.eclipse.aether.collection.CollectResult;
+import org.eclipse.aether.collection.DependencyCollectionContext;
+import org.eclipse.aether.collection.DependencyCollectionException;
+import org.eclipse.aether.collection.DependencyManager;
+import org.eclipse.aether.connector.basic.BasicRepositoryConnectorFactory;
+import org.eclipse.aether.graph.Dependency;
+import org.eclipse.aether.graph.DependencyCycle;
+import org.eclipse.aether.graph.DependencyNode;
+import org.eclipse.aether.graph.DependencyVisitor;
+import org.eclipse.aether.graph.Exclusion;
+import org.eclipse.aether.impl.DefaultServiceLocator;
+import org.eclipse.aether.internal.impl.collect.DefaultDependencyCollectionContext;
+import org.eclipse.aether.repository.LocalRepository;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.resolution.ArtifactDescriptorException;
+import org.eclipse.aether.resolution.ArtifactDescriptorRequest;
+import org.eclipse.aether.resolution.ArtifactDescriptorResult;
+import org.eclipse.aether.spi.connector.RepositoryConnectorFactory;
+import org.eclipse.aether.spi.connector.transport.TransporterFactory;
+import org.eclipse.aether.transport.file.FileTransporterFactory;
+import org.eclipse.aether.transport.http.HttpTransporterFactory;
+import org.eclipse.aether.util.artifact.JavaScopes;
+import org.eclipse.aether.util.graph.manager.DefaultDependencyManager;
+import org.eclipse.aether.util.graph.manager.DependencyManagerUtils;
+import org.eclipse.aether.util.graph.transformer.ConflictResolver;
+import org.eclipse.aether.util.graph.traverser.StaticDependencyTraverser;
+import org.eclipse.aether.util.graph.visitor.TreeDependencyVisitor;
+
+public class MavenResolver implements Resolver {
+
+  private final RemoteRepositoryFactory remoteRepositoryFactory;
+  private final EventListener listener;
+
+  public MavenResolver(Netrc netrc, EventListener listener) {
+    this.remoteRepositoryFactory = new RemoteRepositoryFactory(netrc);
+    this.listener = listener;
+  }
+
+  private Dependency createBom(
+      com.github.bazelbuild.rules_jvm_external.resolver.Artifact artifact) {
+    Coordinates coordinates = artifact.getCoordinates();
+
+    Dependency bom =
+        new Dependency(
+            new DefaultArtifact(
+                coordinates.getGroupId(),
+                coordinates.getArtifactId(),
+                "pom",
+                "",
+                coordinates.getVersion()),
+            JavaScopes.RUNTIME);
+
+    Set<Exclusion> excluded =
```

Comment:
nit: why not use `exclusions` instead of `excluded`

Reply:
Because in my head, I read them as "these are the things being excluded", but `exclusions` works just as well. Changed.